### PR TITLE
[FC-55] Upgrade react and react-dom to 17.0.2

### PR DESCRIPTION
**Pull request by taivueh**
Jira card: [FC-55](https://employmenthero.atlassian.net/browse/FC-55?atlOrigin=eyJpIjoiM2Y4OWVkMDYxNzdiNDc0ZDhiNWY1YjNkM2E5NjBiN2YiLCJwIjoiaiJ9)

Run test suites:

Screenshot 2023-08-22 at 15 10 58

Video demo:

https://github.com/Thinkei/react-signature-pad/assets/116699596/698039a1-0717-4527-9d93-1d9e05af0343

Double test it on hr-web-app:

https://github.com/Thinkei/react-signature-pad/assets/116699596/7f48df72-5905-457a-931c-80dbea522282

[FC-55]: https://employmenthero.atlassian.net/browse/FC-55?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -46,8 +46,8 @@
   },
   "peerDependencies": {
     "prop-types": "^15.5.8",
-    "react": "0.14 - 16",
-    "react-dom": "0.14 - 16"
+    "react": "0.14 - 17.0.2",
```
Comment (EH-SonLe) [inline package.json:49]: it should be `16.9.0 || 17.0.2`, IMO :trollface:

Comment (taivueh) [reply package.json]: yup, let keep it consistence with libs we already updated 👍

**Review by EH-SonLe — APPROVED** [review]

**Review by sangquangnguyen — COMMENTED** [review]
LGTM

**Review by sangquangnguyen — APPROVED** [review]
Queen approve :lol:

File: test-utils/configure-enzyme.js [hunk 2]
```diff
@@ -1,4 +1,4 @@
 import { configure } from 'enzyme';
-import Adapter from 'enzyme-adapter-react-16';
+import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
```
Comment (taivueh) [inline test-utils/configure-enzyme.js:2]: you can found the explaination for this one at here: https://github.com/Thinkei/report-board/pull/366

**Review by christhuong — DISMISSED** [review]

File: test-utils/configure-enzyme.js (hunk 2 above)
Comment (taivueh) [reply test-utils/configure-enzyme.js:2]: tldr: enzyme hasn't support react 17 yet, so I changed it to a different one that support 17 well

**Review by vthanh6494 — APPROVED** [review]

File: package.json [hunk 3]
```diff
@@ -64,6 +64,7 @@
     "css-loader": "^6.8.1",
     "enzyme": "^3.10.0",
     "enzyme-adapter-react-16": "^1.14.0",
```
Comment (EH-SonLe) [inline package.json:66]: should we remove it if unnecessary? :trollface:

Comment (taivueh) [reply package.json:66]: I think so, just test it again and still works fine without `enzyme-adapter-react-16`

**Review by sangquangnguyen — APPROVED** [review]

**Review by EH-SonLe — APPROVED** [review]

**Pull request merged**